src/cmap/connection.ts

```diff
@@ -1181,121 +1186,83 @@ export class ModernConnection extends TypedEventEmitter<ConnectionEvents> {
     };
     exhaustLoop().catch(replyListener);
   }
-}
 
-const kDefaultMaxBsonMessageSize = 1024 * 1024 * 16 * 4;
-
-/**
- * @internal
- *
- * This helper reads chucks of data out of a socket and buffers them until it has received a
- * full wire protocol message.
- *
- * By itself, produces an infinite async generator of wire protocol messages and consumers must end
- * the stream by calling `return` on the generator.
- *
- * Note that `for-await` loops call `return` automatically when the loop is exited.
- */
-export async function* readWireProtocolMessages(
-  connection: ModernConnection,
-  { signal }: { signal?: AbortSignal } = {}
-): AsyncGenerator<Buffer> {
-  const bufferPool = new BufferPool();
-  const maxBsonMessageSize = connection.hello?.maxBsonMessageSize ?? kDefaultMaxBsonMessageSize;
-  for await (const [chunk] of on(connection.socket, 'data', { signal })) {
-    if (connection.delayedTimeoutId) {
-      clearTimeout(connection.delayedTimeoutId);
-      connection.delayedTimeoutId = null;
-    }
-
-    bufferPool.append(chunk);
-    const sizeOfMessage = bufferPool.getInt32();
+  /**
```

Comment:
where did `kDefaultMaxBsonMessageSize` go?

Reply:
Yeah I think we still need this check correct? Maybe the server won't send us something larger but something impersonating the server could.

Reply:
> where did kDefaultMaxBsonMessageSize go?

Apologies I didn't capture this in the PR description, I removed it. I intended to make a follow-up, will do.

> Yeah I think we still need this check correct?

There is something very questionable about this flag:
1. We never respect the value that comes from the server because we [check the hello for the value](https://github.com/mongodb/node-mongodb-native/blob/main/src/cmap/connection.ts#L232), but the hello is [not available until the handshake is performed](https://github.com/mongodb/node-mongodb-native/blob/main/src/cmap/connect.ts#L162).
2. More importantly, `maxBsonMessageSize` is not a field in the hello reply:  https://www.mongodb.com/docs/manual/reference/command/hello/ We seem to have a mixture of `maxBsonObjectSize` and `maxMessageSizeBytes`.
  - `maxBsonObjectSize` and others are the fields used by bulkWrite to split a bulk op into correctly sized chunks
3. `maxMessageSizeBytes` is probably what was intended here but this default exceeds the server default by quite a bit, this is ~67mb the server's default is 48mb. In either case that is the cap on the server's input. 
  - I am not sure as a client we should be validating anything other than what OP_MSG supports which would be int32 max, we also know our BSON library cannot process a document more than 16mb

Happy to reintroduce an upper limit to make us feel "safe" but the actual answer to all of the above confusion may require a bit more research 🤔

Reply:
I'm happy to have this as a follow up. Just wanted to make sure we don't remove it without some research after if we need it.

Reply:
https://jira.mongodb.org/browse/NODE-5800